src/core/multifeaturelistmodelbase.cpp

```diff
@@ -364,10 +364,6 @@ bool MultiFeatureListModelBase::deleteFeature( QgsVectorLayer *layer, QgsFeature
       QgsMessageLog::logMessage( tr( "Cannot rollback layer changes in layer %1" ).arg( layer->name() ), "QField", Qgis::Critical );
   }
 
-  //delete parent
```

Comment:
The actual change, it was already deleted a few lines above. Artifact from the previous "silent" deleting, my bad I haven't cought this, because deleting works, but not the following operations on the layer.